reporting/impl/src/main/java/org/jboss/windup/reporting/rules/generation/techreport/TechReportService.java

```diff
@@ -218,11 +219,23 @@ public TechStatsMatrix getTechStatsMap(ProjectModel onlyForApplication)
             }
             else
             {
-                appsToCountTowards = Collections.singletonList(onlyForApplication.getId());
+                appsToCountTowards = Collections.singletonList(appBeingSummarized.getId());
             }
 
+            boolean isSharedApp = appsToCountTowards.size() > 1;
+            List<ProjectModel> apps = stat.getProjectModel().getApplications();
+
             for (Long appToCountTowards : appsToCountTowards)
             {
+                if(isSharedApp)
+                {
+                    List<ProjectModel>sharedApps = apps.stream().filter(app -> app.getName().toLowerCase().contains("shared")).collect(toList());
+                    if(!sharedApps.stream().anyMatch(app -> app.getElement().id().equals(appToCountTowards)))
```

Comment:
Could we remove getting the sharedApps list and instead do this filter ? What do you think ?

if ( apps.stream().anyMatch(app -> app.getName().toLowerCase().contains("shared") && app.getElement().id().equals(appToCountTowards) )

Reply:
I tried this but it is subtly different to the original and doesn't produce the correct results so I've left it with the sharedApps list retrieval moved outside of the loop and the original stream/filtering code

Reply:
+1